packages/ariakit/src/combobox/combobox-item.tsx

```diff
@@ -73,14 +82,16 @@ export const useComboboxItem = createHook<ComboboxItemOptions>(
 
     const onClickProp = useEventCallback(props.onClick);
 
+    const hideOnClickProp = useBooleanEventCallback(hideOnClick);
+
     const onClick = useCallback(
       (event: MouseEvent<HTMLDivElement>) => {
         onClickProp(event);
         if (event.defaultPrevented) return;
         if (setValueOnClick && value != null) {
           state?.setValue(value);
         }
-        if (hideOnClick) {
+        if (hideOnClickProp(event)) {
```

Comment:
Let's add `hideOnClickProp` to the callback deps just to keep it consistent.

Reply:
Ah, I'm used to the react hooks eslint config screaming at me when I miss these, so I didn't realize it was missing from the deps. Good catch. May I ask why is this rule disabled in this project? Is it because we need finer control over dep arrays?